### PR TITLE
Implementation of FASTALoader

### PR DESCRIPTION
**Pull request by rbharath**
This PR adds a new `FASTALoader` class to `dc.data` as suggested in #1002. As @jisraeli suggests, reuses DRAGONN utilities for loading FASTA files. Note that this PR loads FASTA files into Numpy arrays of shape `(N_sequences, 4, seq_length, 1)` instead of `(N_sequences, 1, 4, seq_length)` as in the original DRAGONN implementation. The reason for this is that Keras/TensorFlow now defaults image handling to have the color channels at the end instead of the front (https://github.com/kundajelab/dragonn/issues/44) so this change will make it easier to process data without reshapes.

Will probably make sense to eventually swap this custom parser out for `biopython`, but that can be left for a future PR.

### CODE REVIEW

File: deepchem/data/data_loader.py [hunk 1]
```diff
@@ -186,7 +187,20 @@ def __init__(self,
...
   def featurize(self, input_files, data_dir=None, shard_size=8192):
-    """Featurize provided files and write to specified location."""
+    """Featurize provided files and write to specified location.
+    
+    For large datasets, automatically shards into smaller chunks
+    for convenience.
+
+    Parameters
+    ----------
+    input_files: list
```
Comment (lilleswing) [inline deepchem/data/data_loader.py]: files or strings?

Comment (lilleswing) [reply deepchem/data/data_loader.py]: its strings

File: deepchem/utils/save.py [hunk 2]
```diff
@@ -104,6 +106,68 @@ def load_csv_files(filenames, shard_size=None, verbose=True):
...
+  # TODO(rbharath): Unlike the DRAGONN implementation from which this
+  # was ported, I couldn't transform the "ACGT..." strings into
+  # integers all at once. Had to do one at a time. Might be worth
+  # figuring out what's going on under the hood.
+  for sequence in sequences:
+    integer_seq = label_encoder.transform(
+        np.array((sequence,)).view(integer_type))
+    integer_array.append(integer_seq)
+  integer_array = np.concatenate(integer_array)
+  integer_array = integer_array.reshape(len(sequences), sequence_length)
```
Comment (lilleswing) [inline deepchem/utils/save.py]: I'm suspicous of this reshape -- can we get a test?

File: deepchem/data/data_loader.py (hunk 1 above)
Comment (rbharath) [reply deepchem/data/data_loader.py]: Good point. Fixed description to state "filenames"

File: deepchem/utils/save.py (hunk 2 above)
Comment (rbharath) [reply deepchem/utils/save.py]: Good catch. Turns out that there was an error case when input sequences had different lengths. Raise a `ValueError` for that now, and added a couple more tests.

**Comment by rbharath** [comment]
@lilleswing Addressed comments. Could you take another look?

**Comment by lilleswing** [comment]
LGTM

**Pull request merged**